### PR TITLE
fail rtd on warning added

### PR DESCRIPTION
**Pull request by yafimvo**
Ensures doc building fails when a notebook fails

Excluded: 'myst.domains', 'myst.header', and 'misc.highlighting_failure'

Failed run of rtd: 
https://readthedocs.org/projects/jupysql/builds/19507324/

Original issue : [ensure doc building fails when a notebook fails](https://github.com/ploomber/contributing/issues/23)

----
:books: Documentation preview :books:: https://jupysql--133.org.readthedocs.build/en/133/

### CODE REVIEW

File: doc/_config.yml [hunk 1]
```diff
@@ -40,6 +40,7 @@ html:
 sphinx:
   config:
     execution_show_tb: true
+    suppress_warnings: ['myst.domains', 'myst.header', 'misc.highlighting_failure']
```
Comment (edublancas) [inline doc/_config.yml:43]: do you remember what these warnings are? it'd be good if we document them with a comment so we know what we're ignoring.

I think highlighting_failure is just when a cell has a programming language that doesn't have the syntax highlighting  but I'm unsure about the first two

**Review by edublancas — CHANGES_REQUESTED** [review]

File: doc/_config.yml (hunk 1 above)
Comment (yafimvo) [reply doc/_config.yml:43]: yes. This is the [failing run](https://readthedocs.org/projects/jupysql/builds/19505971/).

## `myst.header`
Ignores 4 `Non-consecutive header level` - this means we do not use header hierarchy as recommended.
e.g skipping header2
```
# header1

### header3
```
## `myst.domains`
I saw this in our previous `conf.py` file so I added it to the `_conf.py`. this basically should suppress this warning:

```
WARNING: Domain 'sphinxcontrib.bibtex.domain::cite' has not implemented
```

[took it from here](https://github.com/executablebooks/jupyter-book/pull/1227)

**Comment by yafimvo** [comment]
rebased.
Suppressed warnings were added directly to `conf.py` instead of `_conf.py`.

File: doc/_config.yml (hunk 1 above)
Comment (edublancas) [reply doc/_config.yml]: can we fix the warnings instead of suppresing them? looks like the one is easy, unsure about the second one

Comment (yafimvo) [reply doc/_config.yml]: Yes, but we need to keep following these rules when writing docs our docs.

as for the domain error - I'll try to fix it as well.

**Comment by yafimvo** [comment]
@edublancas ready for review

**Comment by yafimvo** [comment]
@edublancas ready for review

**Review by edublancas — APPROVED** [review]

**Pull request merged**